### ISSUE TITLE
IP address and HMAC signature check => Notification has been rejected because the IP address could not be verified

### ISSUE THREAD

**Issue by LittleBigDev** [Adyen/adyen-magento2#780]
**Describe the bug**
"Notification has been rejected because the IP address could not be verified". After investigation, this is because of IP validation. More details in Additional Context section

**To Reproduce**
Prerequisites:
In received notification request headers, `HTTP_X_FORWARDED_FOR` must look like `XX.XX.XX.XX, YY.YY.YY.YY`

1. In admin, enable "Check notification's IP address and HMAC signature"
2. In front, prepare an order to be placed
3. Monitor `var/log/adyen/notification.log` file with a `tail -f` for example
4. Enter credit card info and submit payment form

**Expected behavior**
Regular notification logs in log file

**Actual behavior**
```
[2020-07-31 09:09:03] AdyenLoggerTest.ADYEN_NOTIFICATION: Notification has been rejected because the IP address could not be verified {"is_exception":false} []
```

**Desktop (please complete the following information):**
 - Irrelevant

**Additional context**
Our test server is hosted on AWS EC2. I don't knwow why exactly we have HTTP_X_FORWARDED_FOR with concatenated IPs, but here we are : there are concatenated IPs in this header.

`\Adyen\Payment\Helper\IpAddress::isIpAddressValid()` method is unable to match any IP because passed IP values were not exploded in `\Adyen\Payment\Controller\Process\Json::isIpValid()`.
In `isIpValid()` method, IP addresses retrieved from `$_SERVER['HTTP_X_FORWARDED_FOR']` should be exploded on comma character (`,`), then trimmed, and only then pushed into `$ipAddress` array.

Thank you by advance !

**Comment by AlexandrosMor** [comment]
Hello @LittleBigDev,

Thank for raising this issue. We are going to look into it and come back to you as soon as possible.

regards,
Alexandros

**Comment by acampos1916** [comment]
Hey @LittleBigDev,

This has been addressed in #784, thanks for the feedback.

Ángel

**Comment by LittleBigDev** [comment]
Thank you @acampos1916 

I left a comment there. This is a double bummer for us : your fix doesn't validate IPs from `HTTP_X_FORWARDED_FOR` anymore. And because of that (and our load balancer), we won't be able to enable either IP validation or HMAC key validation :(

Actually, any server architecture hiding real remote address to magento won't be compatible with your IP and HMAC key validation feature :/
